### ISSUE TITLE
Backup files showing up in library

### ISSUE THREAD

**Issue by RandomDennisLee** [janeczku/calibre-web#2120]
**Describe the bug/problem**
Calibre has a feature, 'Polish books', which writes metadata to the book file directly. A backup of the original file is first created under the filename '_bookname_.original__[format]_'. E.g. 'War of the Worlds.epub' gets backed up as 'War of the Worlds.original_epub'. Ideal behaviour would be for Calibre Web to ignore these files. At the moment however, they get listed as available alternate file format of type 'original_format'. Issue also exists in OPDS catalog.

**To Reproduce**
1. Edit metadata for book
2. Polish book
3. See available formats for download in calibre web library 

**Logfile**
n/a

**Expected behavior**
Calibre backup copies should be ignored

**Environment (please complete the following information):**
 - OS: Win 10
 - Python version: 3.9.6
 - Calibre-Web version: 0.6.12
 - Docker container: None
 - Special Hardware: None
 - Browser: Firefox

**Additional context**

**Comment by OzzieIsaacs** [comment]
Which version of calibre are you using?

**Comment by RandomDennisLee** [comment]
5.27 (current latest release)

**Comment by AutoGitr** [comment]
There is a setting that makes Calibre automatically delete the original_epub after polishing. 

Calibre -> Preferences -> Tweaks -> Save original file when converting/ polishing from same format to same format -> Change "save_original_format_when_polishing = True" to "save_original_format_when_polishing = False"

IMO, if you want to keep a backup of your books they shouldn't be stored in calibre. The folder that i import my ebooks from into calibre is basically my backup. Any books in calibre are meant for reading. Then I keep both that folder and the calibre library folder backed up. If this solution works for you changing the setting would be an instant fix.

**Comment by RandomDennisLee** [comment]
It's a valid workaround. However, I see no reason why we wouldn't expect a web UI for Calibre to support native Calibre functionality and instead do it manually.

**Comment by wgibcus** [comment]
> It's a valid workaround. However, I see no reason why we wouldn't expect a web UI for Calibre to support native Calibre functionality and instead do it manually.

I think you might be confused. Calibre-web isn't the web interface for Calibre. Calibre has its own web interface. Calibre-web is designed to be a replacement for the Calibre app. Any functionality of the Calibre app is essentially irrelevant with regards to Calibre-web.

**Comment by RandomDennisLee** [comment]
I said calibre web is **A** web UI for calibre, not **THE**. Also, what point are you trying to make here, with regards to the original issue of unexpected and incorrect behaviour?

**Comment by wgibcus** [comment]
Well there is no reason why this sort of functionality should be included in Calibre-web as Calibre app and Calibre-web are two separate projects. I don't think it's reasonable at all to expect that Calibre-web needs to include functionality from another app, even a related one.

**Comment by RandomDennisLee** [comment]
What functionality are you talking about? I don't know if you've read the issue correctly, but we're not adding any functionality. 

The entire, and only, point of calibre web is to provide a user friendly web interface for calibre. That scope presumably covers NOT pulling invalid files out of the filesystem and offering them to users.

**Comment by wgibcus** [comment]
> It's a valid workaround. However, I see no reason why we wouldn't expect a web UI for Calibre to support native Calibre functionality and instead do it manually.

**Comment by RandomDennisLee** [comment]
I stand corrected. Let's just serve the user a bunch of invalid files then

**Comment by wgibcus** [comment]
> I stand corrected. Let's just serve the user a bunch of invalid files then

"Calibre-Web is a web app providing a clean interface for browsing, reading and downloading eBooks using an existing Calibre database." It does not require Calibre app at all.

**Comment by RandomDennisLee** [comment]
Yes, let's all dump calibre and just create and modify databases by hand. Do you have anything useful to actually contribute here or are you just bored and trolling?

**Comment by wgibcus** [comment]
> Yes, let's all dump calibre and just create and modify databases by hand. Do you have anything useful to actually contribute here or are you just bored and trolling?

You're being quite rude and sarcastic. All I've done is point out the error in your statement. Calibre-web is not directly connected to the Calibre app and is designed to be a replacement for the app, not just be the web server alternative. The fact you choose to use Calibre app and Calibre-web  togetheris entirely on you and the Calibre-web developers have no need to implement any fix for Calibre app issues. Different developers, different purpose for their work. 

Maybe try putting aside your assumptions and read the stated purpose of Calibre-web, which I got from this very site, and understand that just because you rudely demand something, it's going to happen for you.

**Comment by OzzieIsaacs** [comment]
@jeff47

**Comment by Magic815** [comment]
Hi Ozzie - I was hoping to add my voice to the mix here as I think having a feature to hide certain formats (as also mentioned in #2194) would be a really nice thing for Calibre-Web to be able to do.

In addition to the example of hiding "original_epub" files, it could be easily expanded to other file types. For instance, in my personal Calibre library, I add the audiobook (m4b file) to the book listing in Calibre, so it's all in one place. But I don't need to have my audiobooks reachable on the calibre-web site - so it'd be great if I could just set calibre-web to ignore "*.m4b" files.

Maybe it could be as simple as a comma delineated user entry for every file type to ignore? So in my case, it could be ".original_epub, .m4b". Curious to hear your thoughts!
